SPEC_MAINTAINER_PROTOCOL.md

```diff
@@ -0,0 +1,80 @@
+# Spec Maintainer Protocol
+
+> We have a protocol to maintain specs of protocols! How meta, but very useful.
+
+## Motivation
+
+The protocol specifications should be treated as the ultimate source of truth for how the protocol works, how to implemented and what users and builders can expect on how it behaves. However, writting specs for a protocol is not novelty and there has been multiple decades of research on how to do this well, from having a reference implementation, standards bodies and/or formal verification to ensure both code and specs behave the same. It is a hard problem and ultimately one that questions how programming is done as a whole.
+
+Understanding this, we want to make sure to provide a platform in which the Core Developers and the whole IPFS community can collaborate and build specs and code that as much of a pair as possible. This compromises into two main chunks of work:
+
+- Writting the spec and iterating on it to the point that the spec and the code progress smoothly together.
+- For mature and more stable specs, have tests (using [Protocol Driven Development](https://github.com/ipfs/pdd), [TLA+](https://lamport.azurewebsites.net/tla/tla.html) or another framework of verification) to ensure that all implementations stand correct.
+
+The very aspirational future is to have a spec language in which we can describe how the program should function and it will automatically create the tests to verify it. The aspiration future ++ is having such spec language that writes the code itself and verifies it 🚀.
+
+The Spec Maintainer Protocol takes a large amount of inspiration from the repo [Lead Maintainer Protocol](https://github.com/ipfs/team-mgmt/blob/master/LEAD_MAINTAINER_PROTOCOL.md).
```

Comment:
We need to be careful about this (also noted below). Specs are, effectively, append-only logs. They're not code and shouldn't be treated like code.

Reply:
Do not disagree. Any special point you want to highlight as a note of caution?